src/main/org/codehaus/groovy/ast/VariableScope.java

```diff
@@ -178,4 +178,26 @@ public Object removeReferencedClassVariable(String name) {
     public Iterator<Variable> getReferencedClassVariablesIterator() {
         return getReferencedClassVariables().values().iterator();
     }
-}
\ No newline at end of file
+
+    /**
+     * Gets a map containing the variables declared in this scope.
+     * This map cannot be modified.
+     * @return a map containing the declared variable references
+     */
+    public Map<String, Variable> getDeclaredVariables() {
```

Comment:
please do not change it for getReferencedClassVariables, we have code directly mutating that map